File: julia/random.jl
#!/usr/bin/julia


function uniform_random(a, b, seed)
    local t

    seed = 2045 * seed + 1
    #println("seed1:", seed)
    seed = seed - ( seed / 1048576) * 1048576
    #println("seed2:", seed)
    t = seed/1048576.0
    t = a + (b-a) * t

    return t
end

seed=13579

for i=1:10

    #println("seed:",seed)

    x=uniform_random(0, 1.0, seed)
    print(x,'\n')
end

File: lua/random.lua
#!/usr/bin/lua

local seed=13579

function uniform_random(a, b)
    local t

    seed = 2045 * seed + 1
    seed = seed - math.modf(seed/1048576) * 1048576
    t = seed/1048576.0
    t = a + (b-a) * t

    return t
end

for i=0, 10, 1 do
    x=uniform_random(0, 100.0)
    print(x)
end
